x-pack/plugins/apm/server/lib/fleet/get_apm_package_policy_definition.ts

```diff
@@ -190,4 +190,24 @@ export const apmConfigMapping: Record<
     name: 'tls_curve_types',
     type: 'text',
   },
+  // 'apm-server.anonymous_enabled': {
+  //   name: 'anonymous_enabled',
+  //   type: 'bool',
+  // },
+  // 'apm-server.anonymous_allow_agent': {
+  //   name: 'anonymous_allow_agent',
+  //   type: 'text',
+  // },
+  // 'apm-server.anonymous_allow_service': {
+  //   name: 'anonymous_allow_service',
+  //   type: 'text',
+  // },
+  // 'apm-server.anonymous_rate_limit_ip_limit': {
+  //   name: 'anonymous_rate_limit_ip_limit',
+  //   type: 'integer',
+  // },
+  // 'apm-server.anonymous_rate_limit_event_limit': {
+  //   name: 'anonymous_rate_limit_event_limit',
+  //   type: 'integer',
+  // },
```

Comment:
Should we keep these commented out with a TODO / issue for future support or just remove these lines?

Reply:
Ah, thanks for reminding me of that. @simitt should I add the new fields to this mapping as well? it is used when the user migrates to data stream

Reply:
APM Servers < `7.15` will keep sending the `rum.*` fields and no `anonymous.*` fields; APM Server >= `7.15` will either send `rum.*` or `anonymous.*` or both. The apm package template mapping >= `0.4.0` only understands the `anonymous.*` fields. Therefore the logic should be

* map `apm-server.auth.anonymous.rate_limit.event_limit` to `anonymous_rate_limit_event_limit` , if not set then map `apm-server.rum.event_rate.limit` to `anonymous_rate_limit_event_limit`
* map `apm-server.auth.anonymous.rate_limit.ip_limit` to `anonymous_rate_limit_ip_limit`, if not set then map `apm-server.rum.event_rate.lru_size` to `anonymous_rate_limit_ip_limit`
* map `apm-server.auth.anonymous.allow_service` to `anonymous_allow_service`, if not set then map `apm-server.rum.allow_service_names` to `anonymous_allow_service`
* add support for all other `apm-server.auth.*` fields

If this conditional mapping is hard to implement with the mapping logic you have, we could also implement something in the apm-server to map the deprecated fields to the new ones and always send the new fields - let us know if that's prefered (would be fairly straight forward in the server).